src/terrain/generation/land_generator.hpp

```diff
@@ -39,6 +39,83 @@ namespace terrain {
 
 namespace generation {
 
+class JsonToTile {
+ protected:
+    Dim height_;
+    Dim height_variance_;
+    Dim width_;
+    Dim width_variance_;
+
+    const std::set<std::pair<MaterialId, ColorId>> elements_can_stamp_;
+    const MaterialId stamp_material_id_;
+    const ColorId stamp_color_id_;
+
+ public:
+    virtual TileStamp operator()(int current_sub_region) const = 0;
+    virtual size_t num() const = 0;
+
+    JsonToTile(const Json::Value& data);
+
+    static std::set<std::pair<MaterialId, ColorId>>
+    read_elements_can_stamp(const Json::Value& data);
```

Comment:
Give this a better name

Reply:
any sudgestions?

Reply:
Pick your favorite

- `get_stampable_elements`
- `read_stampable_elements`